test/any/confluence/set_ops-test.es6.js

```diff
@@ -0,0 +1,248 @@
+            }
+
+            this.slow = !this.slow;
+            return promise;
+          },
+        ],
+      });
+
+      EvilDAO = foam.lookup('org.chromium.mlang.sink.test.EvilDAO');
+
+      foam.CLASS({
+        package: 'org.chromium.mlang.sink.test',
+        name: 'ErrorDAO',
+        extends: 'foam.dao.ProxyDAO',
+
+        methods: [
+          function select() {
+            return Promise.reject(new Error('ErrorDAO: Error on select()'));
+          },
+          function find() {
+            return Promise.reject(new Error('ErrorDAO: Error on find()'));
+          },
+        ],
+      });
+
+      ErrorDAO = foam.lookup('org.chromium.mlang.sink.test.ErrorDAO');
+    });
+
+    it('should yield full set on SET_MINUS(nullSet)', done => {
+      var set = MDAO.create({of: Num});
+      var nullSet = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+      }
+
+      set.orderBy(Num.ID).select(E.SET_MINUS(nullSet))
+          .then(setMinusSink => setMinusSink.getSink())
+          .then(sink => {
+            var array = sink.a;
+            expect(array.length).toBe(10);
+            for (var i = 0; i < 10; i++) {
+              expect(array[i].id).toBe(i);
+            }
+
+            done();
+          });
+    });
+
+    it('should yield full set on SET_MINUS(nullSet)', done => {
+      var set = MDAO.create({of: Num});
+      var nullSet = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+      }
+
+      set.orderBy(Num.ID).select(E.SET_MINUS(nullSet))
+          .then(setMinusSink=> setMinusSink.getSink())
+          .then(sink => {
+            var array = sink.a;
+            expect(array.length).toBe(10);
+            for (var i = 0; i < 10; i++) {
+              expect(array[i].id).toBe(i);
+            }
+
+            done();
+          });
+    });
+
+    it('should remove all items when sets are identical', done => {
+      var set = MDAO.create({of: Num});
+      var subtrahend = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+        subtrahend.put(mkNum(i));
+      }
+
+      set.select(E.SET_MINUS(subtrahend))
+          .then(setMinusSink => setMinusSink.getSink())
+          .then(sink => {
+            var array = sink.a;
+            expect(array.length).toBe(0);
+
+            done();
+          });
+    });
+
+    it('should remove some items when there is overlap', done => {
+      var set = MDAO.create({of: Num});
+      var subtrahend = MDAO.create({of: Num});
+
+      for (var i = 0; i < 10; i++) {
+        set.put(mkNum(i));
+        if ((i % 2) === 0) {
+          subtrahend.put(mkNum(i));
+        }
+      }
+
+      set.orderBy(Num.ID).select(E.SET_MINUS(subtrahend))
+          .then(setMinusSink =>setMinusSink.getSink())
+          .then(sink => {
+            var array = sink.a;
+            expect(array.length).toBe(5);
```

Comment:
I assume in JS array equality isn't a thing? Because if it is, array === [1,3,5,7,9] feels like a much easier to follow check. Possibly same for the [0..10] case above, though that array is longer.